tests/RestartDialogTestDefs.js

```diff
@@ -0,0 +1,285 @@
+        {
+            funcName: "jqUnit.assertFalse",
+            args: [
+                "RestartDialog: PSP is hidden when close is fired",
+                "{that}.app.psp.model.isShown"
+            ]
+        }, { // dialog shown
+            funcName: "jqUnit.assertTrue",
+            args: [
+                "RestartDialog: Dialog is shown when PSP is closed",
+                "@expand:{that}.app.restartDialog.isShown()"
+            ]
+        }
+    ],
+
+    { // show PSP
+        func: "{that}.app.psp.show"
+    },
+    [
+        /*
+         * Should hide restart dialog when PSP is showed
+         */
+        {
+            funcName: "jqUnit.assertTrue",
+            args: [
+                "RestartDialog: PSP is show when opened normally",
+                "{that}.app.psp.model.isShown"
+            ]
+        }, {
+            funcName: "jqUnit.assertFalse",
+            args: [
+                "RestartDialog: Dialog is hidden when PSP is shown",
+                "@expand:{that}.app.restartDialog.isShown()"
+            ]
+        }
+    ],
+
+    /*
+     * PSP/Dialog restart buttons interaction
+     * (restart dialog events are handled by PSP as the share the mainIPC channels)
+     */
+    { // Simulate RestartLater click
+        func: "{that}.app.psp.events.onRestartLater.fire"
+    },
+    [
+        /*
+         * Should not show dialog when PSP is closed with RestartLater
+         */
+        {
+            funcName: "jqUnit.assertFalse",
+            args: [
+                "RestartDialog: PSP is hidden when onRestart is fired",
+                "{that}.app.psp.model.isShown"
+            ]
+        }, {
+            funcName: "jqUnit.assertFalse",
+            args: [
+                "RestartDialog: Dialog is hidden on RestartLater in PSP",
+                "@expand:{that}.app.restartDialog.isShown()"
+            ]
+        }
+    ],
+
+    { // show PSP
+        func: "{that}.app.psp.show"
+    },
+    { // Simulate RestartNow click
+        func: "{that}.app.psp.events.onRestartNow.fire"
+    },
+    [
+        /*
+         * Should clear (apply) settings and not show dialog
+         * when PSP is closed with RestartNow
+         */
+        {
+            funcName: "jqUnit.assertFalse",
+            args: [
+                "RestartDialog: PSP is hidden when onRestart is fired",
+                "{that}.app.psp.model.isShown"
+            ]
+        }, {
+            funcName: "jqUnit.assertDeepEq",
+            args: [
+                "RestartDialog: Pending changes are cleared on RestartNow in PSP",
+                [],
+                "{that}.app.settingsBroker.model.pendingChanges"
+            ]
+        }, {
+            funcName: "jqUnit.assertFalse",
+            args: [
+                "RestartDialog: Dialog is hidden on RestartNow in PSP",
+                "@expand:{that}.app.restartDialog.isShown()"
+            ]
+        }
+        // TODO check whether setting applied event was fired
```

Comment:
Can you address this TODO here? Or does it require migration to @kaspermarkus 's branch (which perhaps we are already using)